examples/calculating_time_lags.py

```diff
@@ -0,0 +1,120 @@
+"""
+==========================================
+Computing Cross-Correlations and Time Lags
+==========================================
+
+This example shows how to compute cross-correlations
+between light curves and map the resulting time lags,
+those temporal offsets which maximize the cross-correlation
+between the two signals, back to an image pixel.
+This method
+was developed for studying temporal evolution of AIA intensities
+by `Viall and Klimchuk (2012) <https://doi.org/10.1088/0004-637X/753/1/35>`_.
+The specific implementation in this package is described in detail
+in Appendix C of `Barnes et al. (2019) <https://doi.org/10.3847/1538-4357/ab290c>`_.
+"""
+import dask.array
+import matplotlib.pyplot as plt
+import numpy as np
+
+# sphinx_gallery_thumbnail_number = 4
+import astropy.units as u
+
+from sunkit_image.time_lag import cross_correlation, get_lags, max_cross_correlation, time_lag
+
+
+###################################################################
+# Consider two timeseries whose peaks are separated in time by some
+# interval. We will create a toy model with two Gaussian pulses. In
+# practice, this method is often applied to many AIA light curves
+# and is described in detail in
+# `Viall and Klimchuk (2012) <https://doi.org/10.1088/0004-637X/753/1/35>`_.
+def gaussian_pulse(x, x0, sigma):
+    return np.exp(-((x - x0) ** 2) / (2 * sigma ** 2))
+
+
+time = np.linspace(0, 1, 500) * u.s
+s_a = gaussian_pulse(time, 0.4 * u.s, 0.02 * u.s)
+s_b = gaussian_pulse(time, 0.6 * u.s, 0.02 * u.s)
+plt.plot(time, s_a)
+plt.plot(time, s_b)
+
+###################################################################
+# The problem we are concerned with is how much do we need shift
+# signal A, either forward or backward in time, in order for it to
+# best line up with signal B. In other words, what is the time lag
+# between A and B. In the context of analyzing light curves from
+# AIA, this gives us a proxy for the cooling time between two
+# narrowband channels and thus two temperatures. To find this,
+# we can compute the cross-correlation between the two signals
+# and find which "lag" yields the highest correlation.
+lags = get_lags(time)
+cc = cross_correlation(s_a, s_b, lags)
```

Comment:
Since I am reviewing this first, is there a benefit to doing `cross_correlation(s_a, s_b, lags)` instead of `cross_correlation(s_a, s_b, time)` and calling `get_lags` inside of `cross_correlation`?

Reply:
I went back and forth on this. I think because I wanted to make it explicit that the cross-correlation was a function of `lags` and not `time`. If a user puts in an array of dimension `N` and then gets out a result that has dimension `2N-1`, it's potentially confusing.